tests/unit/test_cli.py

```diff
@@ -211,3 +214,63 @@ def testReferenceBasesListArguments(self):
         self.assertEqual(args.end, 2)
         self.assertEquals(args.outputFormat, "fasta")
         self.assertEquals(args.runner, cli.ListReferenceBasesRunner)
+
+
+class TestOutputFormats(unittest.TestCase):
+    """
+    Tests the different output formats of the cli
+    """
+    class FakeArgs(object):
+        def __init__(self, outputFormat='text'):
+            self.outputFormat = outputFormat
+            self.id = 'id'
+            self.key = 'key'
+            self.baseUrl = 'baseUrl'
+            self.verbose = 'verbose'
+
+    class FakeObject(protocol.ProtocolElement):
+
+        __slots__ = ['id', 'name']
+
+        def __init__(self):
+            self.id = 'id'
+            self.name = 'name'
+
+    def _getRunPrintMethodCalls(self, runner):
+        printCalls = []
+        with mock.patch('__builtin__.print') as printMethod:
+            printMethod.side_effect = \
+                lambda *args, **kwargs: printCalls.append((args, kwargs))
+            runner.run()
+        return printCalls
+
+    def testListReferenceBasesFasta(self):
+        args = self.FakeArgs('fasta')
```

Comment:
two reasons:
1) that's more integration test-y than unit test-y ... mocks make it easy to isolate a particular piece of code and not test any other coupled pieces of code (for instance, if the simulated backend breaks, your version of the test would fail)
2) it would be difficult to capture the output without invoking client_dev.py in another process, since run calls directly to print